sklearn/gaussian_process/kernels.py

```diff
@@ -1320,9 +1320,12 @@ def __call__(self, X, Y=None, eval_gradient=False):
                 D = squareform(dists**2)[:, :, np.newaxis]
 
             if self.nu == 0.5:
-                K_gradient = K[..., np.newaxis] * D \
-                    / np.sqrt(D.sum(2))[:, :, np.newaxis]
-                K_gradient[~np.isfinite(K_gradient)] = 0
+                K_gradient = np.zeros_like(D)
```

Comment:
Yeah, I don't like the solution that I wrote, but I wasn't sure if using the `with warnings.catch_warnings(): warnings.simplefilter` block was encouraged.